bundles/org.openhab.binding.novafinedust/src/main/java/org/openhab/binding/novafinedust/internal/SDS011Handler.java

```diff
@@ -116,59 +120,88 @@ public void initialize() {
+                    "Communicator instance is null in initializeCommunicator()");
             return;
         }
 
         boolean initSuccessful = false;
-        try {
-            initSuccessful = localCommunicator.initialize(mode, interval);
-        } catch (final IOException ex) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, "I/O error!");
-            return;
-        } catch (PortInUseException e) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, "Port is in use!");
-            return;
-        } catch (TooManyListenersException e) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
-                    "Cannot attach listener to port!");
-            return;
-        } catch (UnsupportedCommOperationException e) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
-                    "Cannot set serial port parameters");
-            return;
+        int retryInit = 3;
+        int retryCount = 0;
+        // sometimes the device is a little difficult and needs multiple configuration attempts
+        while (!initSuccessful && retryCount < retryInit) {
+            logger.trace("Trying to initialize device attempt={}", retryCount);
+            initSuccessful = doInit(localCommunicator, mode, interval);
+            retryCount++;
         }
 
         if (initSuccessful) {
             lastCommunication = ZonedDateTime.now();
             updateStatus(ThingStatus.ONLINE);
 
             if (mode == WorkMode.POLLING) {
-                pollingJob = scheduler.scheduleWithFixedDelay(() -> {
+                dataReadJob = scheduler.scheduleWithFixedDelay(() -> {
                     try {
                         localCommunicator.requestSensorData();
                     } catch (IOException e) {
                         updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
                                 "Cannot query data from device");
                     }
                 }, 2, config.pollingInterval, TimeUnit.SECONDS);
+            } else {
+                // start a job that reads the port until data arrives
+                int reportingReadStartDelay = 10;
+                int startReadBeforeDataArrives = 5;
+                long readReportedDataInterval = (config.reportingInterval * 60) - reportingReadStartDelay
+                        - startReadBeforeDataArrives;
+                logger.trace("Scheduling job to receive reported values");
+                dataReadJob = scheduler.scheduleWithFixedDelay(() -> {
+                    try {
+                        localCommunicator.readSensorData();
+                    } catch (IOException e) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                                "Cannot query data from device");
```

Comment:
You should include the exception message in the status detail so that users have a better idea of what went wrong.